[LeongWZ] iP

LeongWZ opened the pull request:
# MummyPro

> “Your mind is for having ideas, not holding them.” – David Allen [(source)](https://dansilvestre.com/productivity-quotes)

MummyPro frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [X] Managing tasks
- [X] Managing deadlines
- [ ] Reminders (coming soon)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

LeongWZ committed:
Implement Level 0. Rename, Greet, Exit

LeongWZ committed:
Implement Level 1. Echo

LeongWZ committed:
Implement Level 2. Add, List

LeongWZ committed:
Remove duplicate classes

LeongWZ committed:
Implement Run command

LeongWZ committed:
Level 3. Mark as Done

LeongWZ committed:
Implement Level 4. ToDos, Events, Deadlines

LeongWZ committed:
Implement A-TextUiTesting

LeongWZ committed:
Implement Level 5. Handle Errors

LeongWZ committed:
Implement Level 6. Delete

LeongWZ committed:
Implement Use enums

LeongWZ committed:
Make listen method private

LeongWZ committed:
Rename task variables

LeongWZ committed:
Make Task class abstract

LeongWZ committed:
Implement Level 7. Save

LeongWZ committed:
Merge branch 'branch-level-7'

LeongWZ committed:
Clean up java class files

LeongWZ committed:
Merge branch 'branch-level-8'

LeongWZ committed:
Merge branch 'master' into branch-level-7

LeongWZ committed:
Implement Level-8

LeongWZ committed:
Merge branch 'branch-level-8'

LeongWZ committed:
Restore changes from local history

LeongWZ committed:
Merge branch-Level-7

LeongWZ committed:
Restore files from local history

LeongWZ committed:
Merge branch 'master' into A-MoreOOP

LeongWZ committed:
Implement TaskList and Storage classes

LeongWZ committed:
Implement Parser class

LeongWZ committed:
Implement Ui class

LeongWZ committed:
Merge branch 'A-MoreOOP'

LeongWZ committed:
Implement A-Packages

Divide classes into packages.

LeongWZ committed:
Merge branch 'A-Packages'

LeongWZ committed:
Merge branch 'add-gradle-support' of github.com:LeongWZ/ip into add-gradle-support

LeongWZ committed:
Merge branch 'add-gradle-support'

LeongWZ committed:
Implement A-Gradle

Automate project builds using Gradle.

LeongWZ committed:
Implement abstract Command class

LeongWZ committed:
Implement A-JUnit

Add JUnit tests to test the behavior of the code.

LeongWZ committed:
Fix program not blocking for user input

LeongWZ committed:
Add find command

LeongWZ committed:
Implement A-JavaDoc

Add JavaDoc comments to the non private methods.

LeongWZ committed:
Merge branch 'branch-A-JavaDoc'

LeongWZ committed:
Fix non-javadoc checkstyle errors

LeongWZ committed:
Merge branch 'branch-A-CodingStandard'

LeongWZ committed:
Fix javadoc related checkstyle errors

LeongWZ committed:
Add javadoc for FindCommand

LeongWZ committed:
Merge branch-Level-9

LeongWZ committed:
Fix checkstyle errors from merging branch-Level-9

LeongWZ committed:
Fix javadoc and actual arguments mismatch

LeongWZ committed:
Merge branch 'master' into branch-Level-9

LeongWZ committed:
Reimplement TaskList filter method to use predicate

LeongWZ committed:
Merge branch 'branch-Level-9'

WinstonJin commented on src/main/java/Echo.java:
```diff
@@ -0,0 +1,20 @@
+
+    @Override
+    public String toString() {
+        String dashes = "-".repeat(100);
```
I like that you used the repeat(100) instead of typing out - a hundred times

WinstonJin commented on src/main/java/Mummy.java:
```diff
@@ -0,0 +1,194 @@
+      + "| |\\/| | | | | '_ ` _ \\| '_ ` _ \\| | | |\n"
+      + "| |  | | |_| | | | | | | | | | | | |_| |\n"
+      + "|_|  |_|\\__,_|_| |_| |_|_| |_| |_|\\__, |\n"
+      + "                                  |___/ \n";
```
Awesome ASCII Art

WinstonJin commented on src/main/java/Mummy.java:
```diff
@@ -0,0 +1,194 @@
+        command = Command.UNKNOWN;
+      }
+
+      try {
```
I suggest adding some additional classes such as Command Class and its child classes of AddCommand, ExitCommand etc. This is because the Mummy class has too many methods in its body that can be separated

WinstonJin reviewed (commented):
Looks pretty good just that maybe can separate the Mummy class into smaller classes like Command and Parser classes

LeongWZ committed:
Merge branch 'master' into first-pull-request

jessica2828 commented on src/main/java/mummy/command/Command.java:
```diff
@@ -0,0 +1,156 @@
+ * Each command is responsible for executing a specific action and
+ * can be extended to implement different types of commands.
+ */
+public abstract class Command {
```
Good use of OOP by implementing abstract Command class!

jessica2828 commented on src/main/java/mummy/task/TaskList.java:
```diff
@@ -0,0 +1,176 @@
+
+    private final String label;
+
+    private final ArrayList<Task> tasks;
```
Good naming of variables

brendanng7 replied:
I agree, tasks which is a collection of the Task class is named very appropriately.

jessica2828 commented on src/main/java/mummy/task/TaskList.java:
```diff
@@ -0,0 +1,176 @@
+     *
+     * @return The number of tasks in the task list.
+     */
+    public int getCount() {
```
Perhaps countTask() would be a better name? As it is more descriptive and it's not confused with getter methods which are usually used to access the class fields?

jessica2828 commented on src/test/java/mummy/command/CommandTest.java:
```diff
@@ -0,0 +1,99 @@
+    }
+
+    @Test
+    public void getArgumentsShouldReturnArguments() {
```
Perhaps you can rewrite this as getArguments_shouldReturnArguments() following the convention for tests? I think it would make the method name more readable. Similar for the other test methods too

jessica2828 commented on src/main/java/mummy/ui/Ui.java:
```diff
@@ -0,0 +1,80 @@
+ */
+public class Ui {
+
+    private static final String DIVIDER = "-".repeat(100);
```
Nice use of constant with appropriate naming! A good idea to use repeat(100) too 👍

jessica2828 reviewed (commented):
Overall LGTM, perhaps just a bit of nits to pay attention to :)

brendanng7 commented on src/main/java/mummy/ui/Mummy.java:
```diff
@@ -0,0 +1,68 @@
+ * and handling user commands.
+ */
+public class Mummy {
+    private static final String LOGO = " __  __\n"
+            + "|  \\/  |_   _ _ __ ___  _ __ ___  _   _\n"
+            + "| |\\/| | | | | '_ ` _ \\| '_ ` _ \\| | | |\n"
+            + "| |  | | |_| | | | | | | | | | | | |_| |\n"
+            + "|_|  |_|\\__,_|_| |_| |_|_| |_| |_|\\__, |\n"
+            + "                                  |___/ \n";
```
I like your ASCII Art!

brendanng7 commented on src/main/java/mummy/utility/Parser.java:
```diff
@@ -0,0 +1,58 @@
+     * @param defaultValue the default value to be returned if parsing fails
+     * @return the parsed integer value or the default value if parsing fails
+     */
+    public static int parseIntOrDefault(String s, int defaultValue) {
```
I like the naming of this method as it explains what it does nicely

brendanng7 commented on src/main/java/mummy/task/TaskList.java:
```diff
@@ -0,0 +1,176 @@
+     * @param index The index of the task to be marked as done.
+     * @throws TaskListException If the index is out of bounds.
+     */
+    public void markTask(int index) throws TaskListException {
```
I like the naming style for the methods of this class. Very clear what it does

brendanng7 commented on src/main/java/mummy/task/TaskList.java:
```diff
@@ -0,0 +1,176 @@
+ */
+public class TaskList {
+
+    private static final String DEFAULT_LABEL = "Here are the tasks in your list:";
```
I like that you followed the naming conventions for constants here

brendanng7 reviewed (commented):
Overall, I think it's really well done and I found how you parsed the commands through the parents command class very interesting. LGTM!

LeongWZ committed:
Make Command's of factory method accept full command string

yu-sutong reviewed (commented):
All follow coding standard. Well done!

LeongWZ committed:
Implement Level 10 GUI

Add a GUI to the chatbot. Use the JavaFX technology to implement the GUI.

LeongWZ committed:
Implement exit GUI program upon receiving bye command

Upon receiving bye command, exit GUI program after 500 ms.

LeongWZ committed:
Move Parser class to command package

LeongWZ committed:
Resolve checkstyle errors

LeongWZ committed:
Make child command classes final

LeongWZ committed:
Rename test methods correctly

Rename test methods according to the coding standard.

Format of test method's name is either:
<featureToBeTested>_<scenerio>_<expectedResult>
or:
<featureToBeTested>_<expectedResult>

LeongWZ committed:
Merge branch 'branch-A-Checkstyle'

LeongWZ committed:
Merge branch 'branch-Level-10'

LeongWZ committed:
Merge branch 'master' into branch-A-Checkstyle

LeongWZ committed:
Resolve checkstyle errors after branch merge

Resolve checkstyle errors after merging branch-Level-10.

LeongWZ committed:
Merge branch 'branch-A-Checkstyle'

LeongWZ committed:
Remove text-ui-test folder

LeongWZ committed:
Merge branch 'master' into first-pull-request

LeongWZ committed:
Change User and Mummy pictures

LeongWZ committed:
Merge branch 'master' into first-pull-request

LeongWZ committed:
Refactor add task commands

Added AddCommand abstract class for add task commands to inherit from.
AddCommand has addTask method for child classes to add task to the
task list. addTask method is final and cannot be overriden.

Removed addTask method from Command class as the behaviour for adding
tasks falls under AddCommand class.

LeongWZ committed:
Use assertions

Use assert feature (not JUnit assertions) to document important assumptions
that should hold at various points in the code.

LeongWZ committed:
Merge branch 'branch-A-CodeQuality'

LeongWZ committed:
Merge branch 'branch-A-Assertions'

LeongWZ committed:
Set up CI

LeongWZ committed:
Merge branch 'branch-A-CI'

LeongWZ committed:
Merge branch 'master' into first-pull-request

LeongWZ committed:
Implement C-Undo

Provide a way to undo the most recent command.

LeongWZ committed:
Merge branch 'branch-C-Undo'

LeongWZ committed:
Merge branch 'master' into first-pull-request

LeongWZ committed:
Merge branch 'master' into branch-A-CI

LeongWZ committed:
Fix gradlew error on github

LeongWZ committed:
Run chmod in github actions to make gradlew executable

LeongWZ committed:
Fix CI

LeongWZ committed:
Fix CI

LeongWZ committed:
Fix CI

LeongWZ committed:
Fix non-existent file when saving to storage

LeongWZ committed:
Merge branch 'branch-A-CI'

LeongWZ committed:
Merge branch 'master' into first-pull-request

LeongWZ committed:
Implement CommandManager class

LeongWZ committed:
Rectify checkstyle errors

LeongWZ committed:
Rectify CI errors

LeongWZ committed:
Merge branch 'master' into branch-C-Undo

LeongWZ committed:
Fix merge conflict

LeongWZ committed:
Merge pull request #2 from LeongWZ/branch-C-Undo

Implement Undo and Redo commands

LeongWZ committed:
Merge branch 'master' of github.com:LeongWZ/ip

LeongWZ committed:
Merge branch 'master' into branch-A-Assertions

LeongWZ committed:
Create commit for PR

LeongWZ committed:
Merge branch 'master' into branch-A-CodeQuality

LeongWZ committed:
Create commit for PR

LeongWZ committed:
Rectify checkstyle errors

LeongWZ committed:
Merge pull request #3 from LeongWZ/branch-A-CodeQuality

Create PR for branch-A-CodeQuality

LeongWZ committed:
Merge pull request #4 from LeongWZ/branch-A-Assertions

Create PR for branch-A-Assertions

LeongWZ committed:
Merge branch 'master' of github.com:LeongWZ/ip

LeongWZ committed:
Merge branch 'master' into branch-A-UserGuide

LeongWZ committed:
Update README

LeongWZ committed:
Merge pull request #5 from LeongWZ/branch-A-UserGuide

Update README and add Ui.png

LeongWZ committed:
Merge branch 'master' of github.com:LeongWZ/ip

LeongWZ committed:
Rectify typo in README

LeongWZ committed:
Update README.md

LeongWZ committed:
Merge branch 'master' of github.com:LeongWZ/ip

LeongWZ committed:
Merge branch 'master' into first-pull-request
